src/main/java/org/opentripplanner/routing/impl/ATLFareServiceImpl.java

```diff
@@ -0,0 +1,391 @@
+                float newCost = 0;
+                if (defaultFare > lastFareWithTransfer) {
+                    newCost = defaultFare - lastFareWithTransfer;
+                }
+                fare.addFare(fareType, getMoney(currency, newCost));
+                lastFareWithTransfer = defaultFare;
+                return true;
+            } else if (transferClassification.type.equals(TransferType.TRANSFER_WITH_UPCHARGE)) {
+                fare.addFare(fareType, getMoney(currency, (float) transferClassification.upcharge / 100));
+                lastFareWithTransfer = defaultFare;
+                return true;
+            }
+            return true;
+        }
+
+        public float getTotal() {
+            int total = 0;
+            for (Fare f : fares) {
+               total += f.getFare(fareType).getCents();
+            }
+            return (float) total / 100;
+        }
+    }
+
+    /**
+     * Get the ride price for a single leg. If testing, this class is being called directly so the required agency cash
+     * values are not available therefore the default test price is used instead.
+     */
+    private float getRidePrice(Ride ride, Fare.FareType fareType, Collection<FareRuleSet> fareRules) {
+        if (IS_TEST) {
+            // Testing, return default test ride price.
+            if (ride.routeData.getShortName().equalsIgnoreCase("101")) {
+                return DEFAULT_TEST_RIDE_PRICE + 1;
+            } else if (ride.routeData.getShortName().equalsIgnoreCase("102")) {
+                return DEFAULT_TEST_RIDE_PRICE + 2;
+            } else if (ride.routeData.getShortName().equalsIgnoreCase("atlsc")) {
+                return DEFAULT_TEST_RIDE_PRICE - 1;
+            } else if (ride.routeData.getShortName().equalsIgnoreCase("BLUE")) {
+                return 0; // free circulator
+            }
+            return DEFAULT_TEST_RIDE_PRICE;
+        }
+        return calculateCost(fareType, Lists.newArrayList(ride), fareRules);
+    }
+
+    private static class TransferMeta {
+        public final TransferType type;
+        public final int upcharge;
+        public final boolean payOnExit;
+
+        /**
+         * Create a TransferMeta
+         * @param type Type of transfer
+         * @param upcharge Upcharge for the transfer in cents
+         * @param payOnExit Whether the fare is charged at end of leg
+         */
+        public TransferMeta(TransferType type, int upcharge, boolean payOnExit) {
+            this.type = type;
+            this.upcharge = upcharge;
+            this.payOnExit = payOnExit;
+        }
+
+        public TransferMeta(TransferType type) {
+            this(type, 0, false);
+        }
+    }
+
+    private static RideType classify(Ride ride) {
+        Route routeData = ride.routeData;
+        String shortName = routeData.getShortName();
+
+        switch(routeData.getId().getAgencyId()) {
+            case COBB_AGENCY_ID:
+                if (shortName.equalsIgnoreCase("100") ||
```

Comment:
I think this can be made more readable by putting the route short names into a `Set<String>` and using `contains()`.

So this:

```java
Set<String> COBB_EXPRESS_SHORT_NAMES = Set.of("100", "101", "102"); // not sure if this is possible with Java 8
COBB_EXPRESS_SHORT_NAMES.contains(shortName)
```

Reply:
For extra clarity, you could also add the route names directly to the `RideType` instances so it could be
```java
RideType.COBB_EXPRESS.routeNames.contains(shortName)
```

Reply:
And perhaps lowercase everything so that you can get rid of the `equalsIgnoreCase`.

Reply:
Great call! I moved the route names into the enum as you suggested. I didn't even know that was possible, thank you for the suggestion.

Reply:
I've made two suggestion on handling the lower casing.